README.md

```diff
@@ -297,60 +281,209 @@ The complete list of transformations supported and their usage in ImageKit can b
                 // Handle Success Response
             case .failure(let error as UploadAPIError):
                 // Handle Upload Error
             case .failure(let error):
                 // Handle Other Errors
       }
-  }
+    }
 )
 ```
 
+### UploadPolicy
+The `UploadPolicy` class represents a set of conditions that need to be met for an upload request to be executed.
+
+`UploadPolicy.Builder` class is responsible for building the UploadPolicy instances. This class provides following methods to access and modify the policy parameters:
+
+| Parameter | Type | Description  |
+|:---------------------------------------------------|:---------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| requireNetworkType(type: NetworkType) | UploadPolicy.Builder | Specifies the network type required for the upload request. Possible values are `UploadPolicy.NetworkPolicy.ANY` and `UploadPolicy.NetworkPolicy.UNMETERED`. Defaults to `NetworkPolicy.ANY`.                                    |
+| requiresBatteryCharging(requiresCharging: Bool) | UploadPolicy.Builder | Sets whether the device needs to be connected to a charger for the upload request. Defaults to `false`. |
+| setMaxRetries(count: Int) | UploadPolicy.Builder | Sets the maximum number of retries for the upload request. Negative value will throw an `IllegalArgumentException`. Defaults to 5. |
+| setRetryBackoff(interval: Long, policy: UploadPolicy.BackoffPolicy) | UploadPolicy.Builder | Sets the backoff interval in milliseconds and policy (`UploadPolicy.BackoffPolicy.LINEAR` or `UploadPolicy.BackoffPolicy.EXPONENTIAL`) for retry attempts. Defaults to interval of 1000ms and policy of `BackoffPolicy.linear`. This increases the gap between each upload retry in either linear or exponential manner. E. g. if the `interval` is set to 3 seconds, then delay for each retry (n) will increase as following: <ul><li>With linear backoff: 3, 6, 9, 12 seconds.. (`interval` * n).</li><li>With exponential backoff: 3, 6, 12, 24 seconds.. (`interval` * 2<sup>n-1</sup>).</li></ul> |
+
+Example code
+```swift
+let policy = UploadPolicy.Builder()
+    .requireNetworkType(UploadPolicy.NetworkType.UNMETERED)
+    .requiresCharging(true)
+    .setMaxRetries(5)
+    .setRetryBackoff(60000, UploadPolicy.BackoffPolicy.EXPONENTIAL)
+    .build()
+```
+
+## Upload preprocessing
+### Image preprocessing
+The `ImageUploadPreprocessor` class encapsulates a set of methods to apply certain transformations to an image before uploading. This will create a copy of the selected image, which will be transformed as per the given parameters before uploading.
+
+`ImageUploadPreprocessor.Builder` class is responsible for building the ImagePreprocess instances. This class provides following methods to access and modify the policy parameters:
```

Comment:
`ImageUploadPreprocessor.Builder class is responsible for building the UploadPreprocessor instances. `

The context of this line is relevant for android not ios

Reply:
Fixed the class reference